app/routers/provider.py

```diff
@@ -44,6 +54,60 @@ async def read_own_provider(
         raise HTTPException(status_code=404, detail="Provider not found")
     return db_provider
 
+# Return provider details by ID
+@router.get("/{provider_id}", response_model=ProviderResponseDetail)
+async def get_provider_details(
+    provider_id: UUID,
+    supabase_user_id: UUID = Depends(get_current_user_id),
+    session: Session = Depends(get_session)
+):
+
+    provider = session.exec(
+        select(Provider).where(Provider.id == provider_id)
+    ).first()
```

Comment:
We should use the crud_helper function here
```suggestion
    provider = get_one(session, Provider, provider_id)
```